File: 0392-is-subsequence/0392-is-subsequence.c
bool isSubsequence(char* s, char* t) {
    int sLen = strlen(s);
    int tLen = strlen(t);
    int j = 0;
    
    for (int i = 0; i < sLen; i++) {
        bool found = false;
        
        while (j < tLen) {
            if (s[i] == t[j]) {
                j++;
                found = true;
                break;
            }
            j++;
        }
        
        if (!found) {
            return false;
        }
    }
    
    return true;
}

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int i=0;
        int j=0;
        while(i<t.length() && j<s.length()){
            char c1=t.charAt(i);
            char c2=s.charAt(j);
            if(c1==c2){
                i++;
                j++;
            }
            else{
                i++;
            }
        }
        return j == s.length();
    }
}